jaxtronomy/LensModel/Profiles/shear.py

```diff
@@ -0,0 +1,222 @@
+
+    def __init__(self):
+        super(ShearGammaPsi, self).__init__()
+
+    @staticmethod
+    def function(x, y, gamma_ext, psi_ext, ra_0=0, dec_0=0):
+        """
+
+        :param x: x-coordinate (angle)
+        :param y: y0-coordinate (angle)
+        :param gamma_ext: shear strength
+        :param psi_ext: shear angle (radian)
+        :param ra_0: x/ra position where shear deflection is 0
+        :param dec_0: y/dec position where shear deflection is 0
+        :return:
+        """
+        # change to polar coordinate
+        r, phi = param_util.cart2polar(x - ra_0, y - dec_0)
+        f_ = 1.0 / 2 * gamma_ext * r**2 * jnp.cos(2 * (phi - psi_ext))
+        return f_
+
+    @staticmethod
+    @jit
+    def derivatives(x, y, gamma_ext, psi_ext, ra_0=0, dec_0=0):
+        # rotation angle
+        gamma1, gamma2 = param_util.shear_polar2cartesian(psi_ext, gamma_ext)
+        return Shear.derivatives(x, y, gamma1, gamma2, ra_0, dec_0)
+
+    @staticmethod
+    @jit
+    def hessian(x, y, gamma_ext, psi_ext, ra_0=0, dec_0=0):
+        gamma1, gamma2 = param_util.shear_polar2cartesian(psi_ext, gamma_ext)
+        return Shear.hessian(x, y, gamma1, gamma2, ra_0, dec_0)
+
+
+class ShearReduced(LensProfileBase):
+    """Reduced shear distortions :math:`\\gamma' = \\gamma / (1-\\kappa)`. This
+    distortion keeps the magnification as unity and, thus, does not change the size of
+    apparent objects. To keep the magnification at unity, it requires.
+
+    .. math::
+        (1-\\kappa)^2) - \\gamma_1^2 - \\gamma_2^ = 1
+
+    Thus, for given pair of reduced shear :math:`(\\gamma'_1, \\gamma'_2)`, an additional convergence term is calculated
+    and added to the lensing distortions.
+    """
+
+    param_names = ["gamma1", "gamma2", "ra_0", "dec_0"]
+    lower_limit_default = {"gamma1": -0.5, "gamma2": -0.5, "ra_0": -100, "dec_0": -100}
+    upper_limit_default = {"gamma1": 0.5, "gamma2": 0.5, "ra_0": 100, "dec_0": 100}
+
+    def __init__(self):
+        super(ShearReduced, self).__init__()
+
+    @staticmethod
+    @jit
+    def _kappa_reduced(gamma1, gamma2):
+        """Compute convergence such that magnification is unity.
+
+        :param gamma1: reduced shear
+        :param gamma2: reduced shear
+        :return: kappa
+        """
+        kappa = 1 - 1.0 / jnp.sqrt(1 - gamma1**2 - gamma2**2)
+        gamma1_ = (1 - kappa) * gamma1
+        gamma2_ = (1 - kappa) * gamma2
+        return kappa, gamma1_, gamma2_
+
+    @staticmethod
+    @jit
+    def function(x, y, gamma1, gamma2, ra_0=0, dec_0=0):
+        """
+
+        :param x: x-coordinate (angle)
+        :param y: y0-coordinate (angle)
+        :param gamma1: shear component
+        :param gamma2: shear component
+        :param ra_0: x/ra position where shear deflection is 0
+        :param dec_0: y/dec position where shear deflection is 0
+        :return: lensing potential
+        """
+        kappa, gamma1_, gamma2_ = ShearReduced._kappa_reduced(gamma1, gamma2)
```

Comment:
try not to access private methods in other classes. Perhaps then re-name _kappa_reduced() to kappa_reduced()